Bottom: c0261de1bd63ed51ac4386bb4cf8cba6bf59a079
Top:    bac2a0672a0ef648d23b62d19d5372cd59225c57
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-12 11:12:22 -0500

Add Gitlab Authentication


---
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index 1baa3f49..a79778ef 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -18,7 +18,8 @@ RUN pip --no-cache-dir install \
     'buildbot-console-view' \
     'buildbot-grid-view' \
     'buildbot-waterfall-view' \
-    'buildbot-www'
+    'buildbot-www' \
+    'ldap3'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 0a3aed76..e5fc5149 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -49,6 +49,20 @@ mn = reporters.MailNotifier(
 c['services'].append(mn)
 
 ####### WEB SERVER
+userInfoProvider = util.LdapUserInfo(
+        uri=os.getenv('LDAP_URI'),
+        bindUser=os.getenv('LDAP_BIND_USER'),
+        bindPw=os.getenv('LDAP_BIND_PW'),
+        accountBase='ou=user,dc=novatech',
+        groupBase='ou=group,dc=novatech',
+        accountPattern='(&(objectClass=inetOrgPerson)(uid=%(username)s))',
+        accountFullName='displayName',
+        accountEmail='mail',
+        groupMemberPattern='(&(objectClass=groupOfNames)(member=%(dn)s))',
+        groupName='cn',
+        accountExtraFields=['employeeType'],
+)
+
 c['www'] = {
     'port': int(os.getenv('BUILDBOT_WEB_PORT',8010)),
     'plugins': {
@@ -56,13 +70,22 @@ c['www'] = {
         'console_view': True,
         'grid_view': True,
         },
-    'auth':util.UserPasswordAuth([("georgem","tgbyhn")]),
+    'auth': util.GitLabAuth(os.getenv('GITLAB_URI'),
+                            os.getenv('GITLAB_CLIENT_ID'),
+                            os.getenv('GITLAB_CLIENT_SECRET'),
+                            userInfoProvider=userInfoProvider),
     'authz':util.Authz(
         allowRules=[
             util.AnyControlEndpointMatcher(role="admins"),
             ],
         roleMatchers=[
-            util.RolesFromUsername(roles=['admins'], usernames=["georgem"]),
+            util.RolesFromUsername(
+                roles=['admins'],
+                usernames=[
+                    "george.mccollister",
+                    "joseph.lutz",
+                    "andrew.cooper",
+                ]),
             ],
         ),
     }
